lib/brine/test_steps.rb

```diff
@@ -3,51 +3,86 @@
 # Steps used to test this library
 # Not loaded by default (except in the tests)
 #
+HTTP_METHOD='GET|POST|PATCH|PUT|DELETE|HEAD|OPTIONS'
+
 class StubResponse
   attr_accessor :body, :status
+
+  def intialize
+    @body = ''
+    @status = 200
+  end
+end
+
+class StubRequest
+  attr_accessor :method, :path, :headers, :body
+
+  def initialize
+    @headers = {}
+  end
+
+  def method=(value)
+    @method=value.downcase.to_sym
+  end
 end
 
-class Store < Faraday::Adapter
-  attr_reader :calls
+class StubBuilder
+  attr_reader :request, :response
+
+  def initialize
+    @request = StubRequest.new
+    @response = StubResponse.new
+  end
 
-  def initialize(app, calls)
-    super(app)
-    @calls = calls
+  def make_response()
```

Comment:
this is likely to be replaced by a Proc property rather than a method as part of allowing it to be overridden for the upcoming delete tests